pkg/auth/providers/azure/azure_actions.go

```diff
@@ -118,79 +113,23 @@ func (ap *azureProvider) testAndApply(actionName string, action *types.Action, r
-// deprecated Azure AD Graph API.
-func (ap *azureProvider) migrateToMicrosoftGraph() error {
-	cfg, err := ap.updateConfigAndValidatePermissions()
-	if err != nil {
-		return err
-	}
-	if err = ap.applyUpdatedConfig(cfg); err != nil {
-		return err
-	}
-	ap.deleteUserAccessTokens()
-	clients.GroupCache.Purge()
-	return nil
-}
-
-func (ap *azureProvider) updateConfigAndValidatePermissions() (*v32.AzureADConfig, error) {
-	cfg, err := ap.getAzureConfigK8s()
-	if err != nil {
-		return nil, err
-	}
-	if !authProviderEnabled(cfg) {
-		return nil, httperror.NewAPIError(httperror.InvalidState, "the Azure AD auth provider is not enabled")
-	}
-
-	updateAzureADEndpoints(cfg)
-
-	// Try to get a new client, which will fetch a new access token and validate its permissions.
-	_, err = clients.NewMSGraphClient(cfg, ap.secrets)
-	if err != nil {
-		return nil, err
-	}
-	return cfg, nil
-}
-
-func (ap *azureProvider) applyUpdatedConfig(cfg *v32.AzureADConfig) error {
-	if cfg.ObjectMeta.Annotations == nil {
-		cfg.ObjectMeta.Annotations = make(map[string]string)
-	}
-	cfg.ObjectMeta.Annotations[GraphEndpointMigratedAnnotation] = "true"
-	return ap.saveAzureConfigK8s(cfg)
-}
-
-// deleteUserAccessTokens attempts to delete all secrets that contain users' access tokens used for working with
-// the deprecated Azure AD Graph API.
-// It is not possible to filter secrets easily by presence of specific key(s) in the data object. The method fetches all
-// Opaque secrets in the relevant namespace and looks at the target key in the data to find a secret that stores a user's
-// access token to delete.
-func (ap *azureProvider) deleteUserAccessTokens() {
-	secrets, err := ap.secrets.ListNamespaced(tokens.SecretNamespace, metav1.ListOptions{FieldSelector: "type=Opaque"})
-	if err != nil {
-		logrus.Errorf("failed to fetch secrets: %v", err)
-		return
-	}
-	// Provider name for Azure AD is the main key on secret data. This allows to identify the secrets to be deleted.
-	const key = Name
-	for _, secret := range secrets.Items {
-		if _, keyPresent := secret.Data[key]; keyPresent {
-			err := ap.secrets.DeleteNamespaced(tokens.SecretNamespace, secret.Name, &metav1.DeleteOptions{})
-			if err != nil {
-				logrus.Errorf("failed to delete secret %s:%s - %v", tokens.SecretNamespace, secret.Name, err)
-			}
+// Get the current config from the database and, if it is enabled and does not have the new auth flow annotation,
+// do not set the annotation on the new config.
+// This covers the case where admins had been using a deprecated Azure AD setup
+// and must migrate to the new auth config setup after upgrading Rancher.
+// But they may not want to migrate immediately, and could still want to reconfigure their existing, deprecated config by changing
+// the Application Secret after rotating it, for example. This way, Rancher allows the old setup to be reconfigured.
+func processAuthConfigOnSetup(current, new *v32.AzureADConfig) {
```

Comment:
Nit: 

Function could have a better name. I think you aren't really processing the config. More you are checking annotations. Maybe something like `migrateNewFlowAnnotation(current, proposed)` might work better?